app/jobs/regular/auto_join_channel_batch.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Jobs
+  class AutoJoinChannelBatch < ::Jobs::Base
+    def execute(args)
+      return "starts_at or ends_at missing" if args[:starts_at].blank? || args[:ends_at].blank?
+      return "End is higher than start" if args[:ends_at] < args[:starts_at]
+
+      channel = ChatChannel.find_by(
+        id: args[:chat_channel_id],
+        auto_join_users: true,
+        chatable_type: 'Category'
+      )
+
+      return if !channel
+
+      query_args = {
+        chat_channel_id: channel.id,
+        start: args[:starts_at],
+        end: args[:ends_at],
+        suspended_until: Time.zone.now,
+        last_seen_at: 3.months.ago,
+        channel_category: channel.chatable_id,
+        mode: UserChatChannelMembership.join_modes[:automatic]
+      }
+
+      records_created = DB.exec(<<~SQL, query_args)
```

Comment:
Im curious how does the `ON CONFLICT` plays with this `records_created` count? Do we get a + 1 on conflict?

Reply:
Good catch. Not sure, will take a look

Reply:
I opted for removing the `ON CONCLICT`. If the membership exists and following is false, it's because the user left the channel. It doesn't make sense to add them again.

Reply:
agreed